tests/integration/tests/test_util/snap.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
```

Comment:
hmmm... this looks REALLY familiar.  😄

Reply:
Maybe because I shamelessly copied it :D